### PR TITLE
[New feature] add maskrcnn

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

Build maskrcnn model

## How was this patch tested?

unit test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskRCNN.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.models.resnet.{Convolution, ResNet, ResNetMask, Sbn}
+import com.intel.analytics.bigdl.models.resnet.ResNet.{DatasetType, ShortcutType}
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+class MaskRCNN(val inChannels: Int,
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskRCNN.scala]: Move this class into `com.intel.analytics.bigdl.models.maskrcnn`

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MaskRCNN.scala]: oki, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      if (useConv) {
+        Sequential()
+          .add(Convolution(nInputPlane, nOutputPlane, 1, 1, stride, stride))
+          .add(Sbn(nOutputPlane))
+      } else {
+        Identity()
+      }
+    }
+
+    def bottleneck(nInputPlane: Int, internalPlane: Int, nOutputPlane: Int,
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: private

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      val m = Sequential()
+        .add(ConcatTable()
+          .add(s)
+          .add(shortcut(nInputPlane, nOutputPlane, stride, useConv)))
+        .add(CAddTable(true))
+        .add(ReLU(true))
+      m
+    }
+
+    def layer(count: Int, nInputPlane: Int, nOutputPlane: Int,
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: private

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: it's already defined in private `buildResNet50`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala (hunk 3 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: it's already defined in private `buildResNet50`

**Comment by wzhongyuan** [comment]
LGTM

**Comment by zhangxiaoli73** [comment]
jenkins pass: http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/899/

**Pull request merged**